### PR TITLE
chore: upgrade to libp2p v2

### PR DESCRIPTION
**Pull request by danisharora099**
This PR upgrades from libp2p 1.x to libp2p v2

Migration notes: https://github.com/libp2p/js-libp2p/blob/main/doc/migrations/v1.0.0-v2.0.0.md#the-libp2ppeer-id-factory-module-has-been-removed

### CODE REVIEW

**Comment by weboko** [comment]
Have no questions to the PR, we definitely need to upgrade `libp2p`. 
But let's postpone till https://github.com/waku-org/js-waku/pull/2135 is merged.

**Comment by danisharora099** [comment]
> It is risky to include into next release without proper dogfooding and we have some projects requiring https://github.com/waku-org/js-waku/pull/2137 and https://github.com/waku-org/js-waku/issues/2158

Tests passing, PR is ready. There shouldn't be anything to dogfood with this internal upgrade, especially considering all tests  now pass.

Merging it would also resolve https://github.com/waku-org/js-waku/issues/2174 @weboko

**Comment by weboko** [comment]
@danisharora099 did you try dogfooding it? I haven't but I'd like to have it done in various contexts before we merge 

Contexts to check:
- how long and good light push functions
- is there any problems with outbound / inbound streams and connections
- are events propagating the same way we expect them: peer connect and disconnect, identify
- etc

Let's do this on Monday and if nothing caught - we can merge.

File: packages/enr/src/enr.spec.ts [hunk 1]
```diff
@@ -416,15 +417,17 @@ describe("ENR", function () {
...
+    let privateKeyUint8Arr: Uint8Array;
 
     beforeEach(async function () {
-      peerId = await createSecp256k1PeerId();
+      const privateKey = await generateKeyPair("secp256k1");
+      privateKeyUint8Arr = privateKey.raw;
+      peerId = peerIdFromPrivateKey(privateKey);
       enr = await EnrCreator.fromPeerId(peerId);
-      privateKey = await getPrivateKeyFromPeerId(peerId);
+      console.log(enr);
```
Comment (weboko) [inline packages/enr/src/enr.spec.ts:430]:
```suggestion

```

File: packages/core/src/lib/filterPeers.spec.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 
 import { filterPeersByDiscovery } from "./filterPeers.js";
 
 describe("filterPeersByDiscovery function", function () {
   it("should return all peers when numPeers is 0", async function () {
-    const peer1 = await createSecp256k1PeerId();
-    const peer2 = await createSecp256k1PeerId();
-    const peer3 = await createSecp256k1PeerId();
+    const [peer1, peer2, peer3] = await Promise.all([
+      generateKeyPair("secp256k1").then(peerIdFromPrivateKey),
```
Comment (weboko) [inline packages/core/src/lib/filterPeers.spec.ts:13]: it seems to be repetitive thing, let's extract into `tests/utils`

File: packages/tests/tests/multiaddr.node.spec.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import type { PeerId } from "@libp2p/interface";
 import type { PeerInfo } from "@libp2p/interface";
-import { CustomEvent } from "@libp2p/interface";
```
Comment (weboko) [inline packages/tests/tests/multiaddr.node.spec.ts:3]: is this change due to `nodejs@20`/browser having `CustomEvent` available?

Comment (danisharora099) [reply packages/tests/tests/multiaddr.node.spec.ts:3]: Indeed. It's now included part of the engine

File: packages/core/src/lib/filterPeers.spec.ts (hunk 2 above)
Comment (danisharora099) [reply packages/core/src/lib/filterPeers.spec.ts:13]: we often also require the `privateKey` so fine to do it this way

**Pull request merged**